quipucords/scanner/scan_data_log.py

```diff
@@ -0,0 +1,188 @@
+    if UUID_CACHE is not None:
+        return UUID_CACHE
+
+    uuid_obj = scan_data_log_state.DatabaseUUID.objects.all().first()
+    if uuid_obj is None:
+        uuid_obj = scan_data_log_state.DatabaseUUID(uuid=uuid.uuid4())
+        uuid_obj.save()
+    UUID_CACHE = uuid_obj.uuid
+
+    return uuid_obj.uuid
+
+
+# This is a potential performance bottleneck since we are touching the
+# database every time we write a record. This isn't necessary - we
+# could get the state of the counter by reading the last few entries
+# of the existing log files on startup. I don't plan to implement that
+# unless we find that this is actually a problem.
+@transaction.atomic()
+def next_sequence_number():
+    """Get the next sequence number and increment the counter."""
+    seq = scan_data_log_state.LatestSequenceNumber.objects.all().first()
+    if seq is None:
+        seq = scan_data_log_state.LatestSequenceNumber(number=0)
+        seq.save()
+
+    number = seq.number
+    seq.number += 1
+    seq.save()
+
+    return number
+
+
+# We use the standard Python loggers when possible, but with two main
+# differences:
+
+# 1. We have a custom log formatter. The standard Python log format is
+#    nice for humans to read, but not nice for machines to parse. So
+#    we have JSONFormatter, which writes log records as plain JSON
+#    dicts that are easy to parse.
+
+# 2. We don't use LogRecord objects. Almost none of their attributes
+#    apply to what we're doing, and there doesn't seem to be any
+#    advantage to using them, so we just pass plain dictionaries
+#    around.
+
+class JSONFormatter(logging.Formatter):
+    """Write input log messages as plain JSON dicts."""
+
+    def format(self, record):
+        """Format a JSON dict as a JSON string."""
+        # StreamHandler will add a '\n' after every record.
+        return json.dumps(record)
+
+
+# We use multiprocessing and not threads, so we need to use a
+# multiprocessing lock.
+class MultiprocessRotatingFileHandler(handlers.RotatingFileHandler):
+    """A RotatingFileHandler that uses a multiprocessing lock."""
+
+    def createLock(self):
+        """Create a lock."""
+        self.lock = mp.RLock()
+
+
+_HANDLER = MultiprocessRotatingFileHandler(
+    filename=settings.SCAN_DATA_LOG_BASENAME,
+    encoding='utf-8',
+    maxBytes=settings.SCAN_DATA_LOG_MAX_BYTES // NUM_LOG_FILES,
+    backupCount=NUM_LOG_FILES)
+_HANDLER.setFormatter(JSONFormatter())
+
+_DRY_RUN = False
+
+
+def log_record(record):
+    """Write a new record to the input log.
+
+    :param record: the record. a Python dict, with JSON-compatible values.
+    """
+    record['database_uuid'] = str(get_database_uuid())
+    record['sequence_number'] = next_sequence_number()
+
+    if not _DRY_RUN:
+        _HANDLER.emit(record)
+
+
+# pylint: disable=too-many-arguments
+def log_fact(host, fact, value, scan_job, scan_task, source):
+    """Write a new fact to the input log.
```

Comment:
input log --> scan data log ; multiple places.

Reply:
Fixed in multiple places, thanks!